mono/sgen/sgen-gc.c

```diff
@@ -1556,10 +1616,14 @@ collect_nursery (const char *reason, gboolean is_overflow, SgenGrayQueue *unpin_
 
 	binary_protocol_collection_begin (gc_stats.minor_gc_count, GENERATION_NURSERY);
 
-	if (sgen_concurrent_collection_in_progress ())
-		object_ops = &sgen_minor_collector.serial_ops_with_concurrent_major;
-	else
-		object_ops = &sgen_minor_collector.serial_ops;
+	if (sgen_concurrent_collection_in_progress ()) {
+		/* FIXME Support parallel nursery collections with concurrent major */
```

Comment:
For now we only support one set of workers running at a given time, either in the context of a major or a minor. Currently concurrent majors run concurrently with minors (the major on the worker, the minor on the gc thread). If we want to have parallel minors during concurrent collections we would probably need to add some form of context to the workers (so we don't create too many threads), and when the parallel minor starts, we suspend the concurrent context and give priority to the parallel minor, and resume the concurrent context when we are done. Also if we are using multiple workers, it would make sense to use all of them, in order to reduce the current pause time.